GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -852,6 +852,13 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
         {
             ThreadHelper.AssertOnUIThread();
 
+            _revisionGraphColumnProvider.RevisionGraphDrawStyle = RevisionGraphDrawStyleEnum.DrawNonRelativesGray;
+
+            // Apply checkboxes changes also to FormBrowse main menu
+            MenuCommands.TriggerMenuChanged();
+
+            FilterChanged?.Invoke(this, new FilterChangedEventArgs(_filterInfo));
+
             if (!CanRefresh)
             {
                 return;
```

Comment:
The longer I think about this, I get to the conclusion that this is a common bug.
This too simple exit does not implement "lazy commands". It **omits** the UI update although the `AppSetting` has already been toggled.

Reply:
`CanRefresh()` is a guard to not run the init simultaneously.
Earlier, the refresh could be called twice during refresh. I believe this is handled now. But if you change a setting while refresh is ongoing, the change will be ignored.

One way to handle this is maybe to make the semaphore global and run the update serialized.
Maybe Debug.Assert when an update still occurs.

Reply:
The guard using `_isRefreshingRevisions` discarded necessary updates. `CanRefresh` mixed it up with the correct `_updatingFilters`.
The (completed) cancellation of the previous ongoing refresh is working well for me (refer to the updated PR description).

I am not a fan of `Debug.Assert` because then a debug build silently exits if run without a debugger attached.